core/src/main/java/de/danoeh/antennapod/core/feed/FeedItem.java

```diff
@@ -489,4 +489,29 @@ public boolean isAutoDownloadable() {
     public String toString() {
         return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
     }
+
+    @Override
+    public int hashCode() {
+        int result = itemIdentifier != null ? itemIdentifier.hashCode() : 0;
+        result = 31 * result + (int) (feedId ^ (feedId >>> 32));
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        FeedItem other = (FeedItem) obj;
+        if (feedId != other.feedId) {
+            return false;
+        }
+        return getIdentifyingValue().equals(other.getIdentifyingValue());
+    }
```

Comment:
I think those methods are no longer needed. Could you please remove them again?

Reply:
I think they are needed for indexOf to work (maybe not but I introduced it with other things t fix an error), alternatively they are definitly needed when using the new flag count because I would just compare if the first object is still the same using equals and than create a notification with the episode count marked as new